src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts

```diff
@@ -312,3 +316,74 @@ export function unique<T>(...arrays: Array<readonly T[]>): T[] {
   }
   return [...set];
 }
+
+interface FloatLimits {
+  positive: {
+    max: number;
+  },
+  negative: {
+    min: number;
+  },
+  emax: number;
+}
+
+/**
+ * Provides an easy way to validate steps in an equation that will trigger a validation error with
+ * constant or override values due to overflow/underflow. Typical call pattern is:
+ * 
+ * const vCheck = new ConstantOrOverrideValueChecker(t, Type.f32);
+ * const c = vCheck.checkedResult(a + b);
+ * const d = vCheck.checkedResult(c * c);
+ * const expectedResult = vCheck.allChecksPassed();
+ */
+export class ConstantOrOverrideValueChecker {
+  #allChecksPassed = true;
+  #floatLimits?: FloatLimits;
+  #quantizeFn: QuantizeFunc<number>;
+
+  constructor(private t: ShaderValidationTest, type: ScalarType) {
+    switch (type) {
+      case Type.f32:
+        this.#quantizeFn = quantizeToF32;
+        this.#floatLimits = kValue.f32;
+        break;
+      case Type.f16:
+        this.#quantizeFn = quantizeToF16;
+        this.#floatLimits = kValue.f16;
+        break;
+      default:
+        this.#quantizeFn = (v: number) => v;
+        break;
+    }
+  }
+
+  quantize(value: number): number {
+    return this.#quantizeFn(value);
+  }
+
+  // Some overflow floating point values may fall into an abiguously rounded scenario, where they
+  // can either round up to Infinity or down to the maximum representable value. In these cases the
+  // test should be skipped, because it's valid for implementations to differ.
+  isAmbiguousOverflow(value: number): boolean {
+    if (!this.#floatLimits ||
+        (value <= this.#floatLimits.positive.max &&
+          value >= this.#floatLimits.negative.min)) {
+          return false;
+    }
+    return Math.abs(value) < Math.pow(2, this.#floatLimits.emax+1);
```

Comment:
I agree in about f64, since number is f64 we are already hitting issues with AF being an f64, if we add a concrete f64 (and presumably move AF to f128) there is going to need to be some major reworking in the CTS to support that.

Reply:
To clarify: Sounds like we don't need to do anything regarding f64 support now, but may need to in the future? Or are there changes that you'd like to see to how this is being handled today?

Reply:
No changes needed at this point (This is just mostly us waxing philospically about future evolutions of WGSL)